index.js

```diff
@@ -0,0 +1,143 @@
+"use strict";
+
+module.exports = {
+  env: {
+    es6: true,
+    node: true,
+    browser: true
+  },
+  parser: "babel-eslint",
+  parserOptions: {
+    ecmaFeatures: {
+      experimentalObjectRestSpread: true,
+      jsx: true
+    },
+    sourceType: "module"
+  },
+  plugins: ["react"],
+  extends: ["wolox"],
+  globals: {
+    __DEV__: true
+  },
+  rules: {
+    "import/default": "error",
+    "import/export": "error",
+    "import/exports-last": "error",
+    "import/extensions": [
+      "error",
+      "never",
+      { js: "never", svg: "always", scss: "always" }
+    ],
+    "import/first": "error",
+    "import/named": "error",
+    "import/namespace": "error",
+    "import/newline-after-import": "error",
+    "import/no-absolute-path": "error",
+    "import/no-extraneous-dependencies": "error",
+    "import/no-mutable-exports": "error",
+    "import/no-named-as-default-member": "error",
+    "import/no-named-default": "error",
+    "import/no-self-import": "error",
+    "import/no-unresolved": "error",
+    "import/no-useless-path-segments": "error",
+    "import/no-webpack-loader-syntax": "error",
+    "import/order": ["error", { "newlines-between": "always" }],
+    "import/prefer-default-export": "off",
+    "jsx-a11y/anchor-is-valid": "error",
+    "react/boolean-prop-naming": "error",
+    "react/button-has-type": "error",
+    "react/default-props-match-prop-types": "error",
+    "react/destructuring-assignment": [
```

Comment:
@mvbattan @WFolini what do you think about this rule?

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/destructuring-assignment.md

@SKOLZ thinks that it's very annoying to have to do this in every method. While I agree, I also think this rule will give us a lot more consistency that we have today in how we use props/state. The problem is the following:

Good scenario:
```
render() {
  const { a, b, c } = this.props; // this is what the rule enforces
  ...
}
```

Bad scenario:
```
const foo = () => {
  const { onlyThingUsed } = this.props; // The rule also enforces this
  ...
}
```

And there's no way to specify a minimum of properties used to enforce the rule. There are 2 open issues for this, but the mantainers don't want to do it: https://github.com/yannickcr/eslint-plugin-react/issues/1731

We could do it ourselves, but for the time being, what do you think? Do we want it or not?

Reply:
The only 'good think' doing the bad scenario is when we'd like to distinguish what props are being modified at the start of the function.
But, IMHO this is annoying and the 'benefit' previously mentioned shouldn't be called a 'benefit' since functions in general shouldn't be too large.

Reply:
I honestly found it impractical. Few were the times that it help me. I think the code is more neat but I do not see much advantage, it becomes a little bit annoying